src/Input/AbstractParamAwareInput.php

```diff
@@ -82,29 +87,26 @@ public function __construct(InputInterface $input, OutputInterface $output, Ques
         }
 
         $question = $inputParam->getQuestion();
-
         $this->modifyQuestion($question);
 
-        if ($value === null && ! $this->input->isInteractive()) {
+        if (
+            ! $this->isParamValueProvided($inputParam, $value)
+            && ! $this->input->isInteractive()
+        ) {
             $value = $inputParam->getDefault();
         }
 
-        if ($value !== null) {
-            $validator = $question->getValidator();
-            if ($validator) {
-                $validator($value);
-            }
-
-            $normalizer = $question->getNormalizer();
-
-            return $normalizer === null ? $value : $normalizer($value);
+        $valueIsArray = (bool) ($inputParam->getOptionMode() & InputOption::VALUE_IS_ARRAY);
```

Comment:
`$valueIsArray` will always be false ? checked like this:

```php
➜  ~ php -a
Interactive shell

php > var_dump((bool) (false & 8));
bool(false)

php > var_dump((bool) (true & 8));
bool(false)

php > var_dump((bool) (1 & 8));
bool(false)

php > var_dump((bool) (0 & 8));
bool(false)
```

ref https://3v4l.org/bhFWC

Reply:
@samsonasik but you forgot to check with the following values:

```php
var_dump((bool) (8 & 8)); // true
var_dump((bool) (9 & 8)); // true
```

These are interesting cases for us, so when parameter is set to have multiple values.

Reply:
@samsonasik Bitwise operators return an integer. In this particular case, `getOptionMode()` will return an integer, and we are using `&` to determine which bits it shares in common with `InputOption::VALUE_IS_ARRAY`, if any. This will return `0` if there are no bits in common, and an integer greater than or equal to `1` if there are. As such, casting to boolean is used to determine if the current option mode includes that bit.

By **default**, none of the various elements includes it. However, you can use `setOptionMode()` to set a different mask of bits, such as `InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY`. In that case, this operation will evaluate to `true`.

A few of the tests I've provided demonstrate this, as they will only work correctly if `VALUE_IS_ARRAY` is in the option mode bitmask.

Reply:
ok 👍